Classes/Service/RoundTrip.php

```diff
@@ -1176,7 +1177,11 @@ public static function backupExtension(Model\Extension $extension, $backupDir)
                 throw new \Exception('Backup directory is not an allowed absolute path: ' . $backupDir);
             }
         } else {
-            $backupDir = PATH_site . $backupDir;
+            if (Environment::isComposerMode()) {
+                $backupDir = Environment::getProjectPath() . '/'. $backupDir;
```

Comment:
Why not using this also for non-composer mode?

Reply:
Good question. At the moment i have no non-composer installation to test ;-) With composer the uploads folder is outside the public folder -> in the project folder. For non-composer i assume it will be in the public folder.

Reply:
You are right depending to https://docs.typo3.org/m/typo3/reference-coreapi/master/en-us/ApiOverview/Environment/Index.html#getpublicpath 
Environment::getProjectPath() should work with composer and non-composer. I tested it with an test instance and i worked. I changed the code.

What confused me a little with my testinstance that the uploads folder where not generated automatically from the TYPO3 installer or Installtool. And on top the extension folder in uploads where not created when installing extension_builder. This was curious.